src/api/libcellml/validator.h

```diff
@@ -59,6 +59,19 @@ class LIBCELLML_EXPORT Validator: public Logger
      */
     void validateModel(const ModelPtr &model);
 
+    /**
+     * @brief Validate the @p model using the CellML 2.0 Specification, including
+     * any imported entities.
+     *
+     * Validate the given @p model and its encapsulated entities using the
+     * CellML 2.0 Specification, after resolving the imports against the given @p baseLocation.
+     * Any errors will be logged in the @c Validator.
```

Comment:
This seems to suggest that the full model hierarchy will be validated, but implementation currently validates the top-level model only and then adds in importer issues (e.g., circular imports) but doesn't seem to validate all the imported models? (maybe still a work in progress?)

Reply:
I don't think we want to be validating all the imported models. We only want to validate the bits we are importing surely?

Reply:
Exactly, but currently we are not validating the imported bits at all, other than issues that arise during the resolution process. At least at an initial look. If this is ready for review, then I can dive deeper into the behaviour to see if it is actually doing what I think it should be...